worldeditadditions_commands/commands/selectors/srect.lua

```diff
@@ -0,0 +1,51 @@
+-- ███████ ██████  ███████  ██████ ████████
+-- ██      ██   ██ ██      ██         ██
+-- ███████ ██████  █████   ██         ██
+--      ██ ██   ██ ██      ██         ██
+-- ███████ ██   ██ ███████  ██████    ██
+-- lua parse_params_srect("10")
+-- local
+function parse_params_srect(params_text)
+  local find, _, sn1, ax1, sn2, ax2, len = params_text:find("([+-]?)([xyz]?)%s*([+-]?)([xyz]?)%s*(%d*)")
```

Comment:
Wow, that's a find call and a half. This works fine - there's no need to change it - but next time try using `worldeditadditions.split()`, which splits a given string on a pattern (e.g. `%s+` for whitespace). Then you can parse each item from the resulting list one at a time, which makes the code more readable.

Here's an example of it in action: <https://github.com/sbrl/Minetest-WorldEditAdditions/blob/master/worldeditadditions_commands/commands/convolve.lua#L14>

Reply:
that `%s*` stuff is the reason `//srect z a 10` doesn't work. I'm already working on a way to fix it that will quite likely use `worldeditadditions.split()` as you suggest

Reply:
Cool!

Reply:
ok this should be fixed in the latest commit.